### PR TITLE
Fix version origin

### PR DESCRIPTION
**Pull request by IshentRas**
#76

### CODE REVIEW

File: providers/openshift_deploy_metrics.rb [hunk 1]
```diff
@@ -52,8 +47,11 @@ def whyrun_supported?
   end
 
   execute 'Deploy Cluster Metrics' do
-    command "#{node['cookbook-openshift3']['openshift_common_client_binary']} new-app --template=metrics-deployer-template --as=system:serviceaccount:openshift-infra:metrics-deployer #{new_resource.metrics_params.map { |opt, value| " -p #{opt}=#{value}" }.join(' ')} -n openshift-infra --config=admin.kubeconfig"
+    command "#{node['cookbook-openshift3']['openshift_common_client_binary']} new-app --template=metrics-deployer-template ${service_account} #{new_resource.metrics_params.map { |opt, value| " -p #{opt}=#{value}" }.join(' ')} -n openshift-infra --config=admin.kubeconfig"
     cwd node['cookbook-openshift3']['openshift_master_config_dir']
+    environment(
+      'service_account' => ose_major_version.split('.')[1].to_i.between?(3, 4) ? '--as=system:serviceaccount:openshift-infra:metrics-deployer' : ''
```
Comment (jperville) [inline providers/openshift_deploy_metrics.rb]: this is clever way to support both versions with slightly different syntax ("--as")

File: providers/openshift_create_master.rb [hunk 2]
```diff
@@ -49,7 +49,8 @@ def whyrun_supported?
         standalone_registry: new_resource.standalone_registry,
         erb_master_named_certificates: named_certificates,
         etcd_servers: new_resource.etcd_servers,
-        masters_size: new_resource.masters_size
+        masters_size: new_resource.masters_size,
+        ose_major_version: node['cookbook-openshift3']['deploy_containerized'] == true ? node['cookbook-openshift3']['openshift_docker_image_version'] : node['cookbook-openshift3']['ose_major_version']
```
Comment (jperville) [inline providers/openshift_create_master.rb]: we should refactor to add helper classes to not repeat these all over the code.

File: providers/openshift_deploy_metrics.rb (hunk 1 above)
Comment (jperville) [inline providers/openshift_deploy_metrics.rb]: same for this one, need helper to tell major versions (the code to match 1.3, 1.4, 3.3, 3.4 is too clever and duplicated)

File: recipes/master_config_post.rb [hunk 3]
```diff
@@ -29,41 +29,41 @@
 end
 
 execute 'Import Openshift Hosted Examples' do
-  command "#{node['cookbook-openshift3']['openshift_common_client_binary']} create -f #{node['cookbook-openshift3']['openshift_common_hosted_base']} --recursive --config=admin.kubeconfig -n openshift || #{node['cookbook-openshift3']['openshift_common_client_binary']} replace -f #{node['cookbook-openshift3']['openshift_common_hosted_base']} --recursive --config=admin.kubeconfig -n openshift"
+  command "#{node['cookbook-openshift3']['openshift_common_client_binary']} create -f #{node['cookbook-openshift3']['openshift_common_hosted_base']} --config=admin.kubeconfig -n openshift || #{node['cookbook-openshift3']['openshift_common_client_binary']} replace -f #{node['cookbook-openshift3']['openshift_common_hosted_base']} --config=admin.kubeconfig -n openshift"
```
Comment (jperville) [inline recipes/master_config_post.rb]: if recursive flag is removed (v1.2.x compatibility) how are all loaded created from the template directory? IIRC "oc create -f" only accepts a single argument which is a file, unless recursive specified

File: templates/default/scheduler.json.erb [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "apiVersion": "v1", 
     "kind": "Policy", 
     "predicates": [
-<% if @ose_major_version.split('.').last.to_i.between?(1,2) %>
+<% if @ose_major_version.split('.')[1].to_i.between?(1,2) %>
```
Comment (jperville) [inline templates/default/scheduler.json.erb]: helper much needed for these version checks.

File: recipes/master_config_post.rb (hunk 3 above)
Comment (IshentRas) [inline recipes/master_config_post.rb]: It also accepts directories (one depth), recursive go deeper...

**Comment by IshentRas** [comment]
@jperville Agree :+1: 
Starting that part very soon. I believe we need a cut as too much codes...

File: recipes/master_config_post.rb (hunk 3 above)
Comment (jperville) [inline recipes/master_config_post.rb]: ooooh I didn't know

**Pull request merged**